src/index.js

```diff
@@ -1,4 +1,89 @@
-// $(document).ready(function(){
+$(document).ready(function(){
+//selecting html elements for use in color change functions
+document.getElementById("name").onkeydown = function() {nameInputBackgroundChange()};
+document.getElementById("mail").onkeydown = function() {mailInputBackgroundChange()};
+document.getElementById("subject").onkeydown = function() {subjectInputBackgroundChange()};
+document.getElementById("msg").onkeydown = function() {msgInputBackgroundChange()};
+
+//selecting html elements for use in reverting colors back to original color upon focus change
+document.getElementById("name").onchange = function() {nameInputChange()};
+document.getElementById("mail").onchange = function() {mailInputChange()};
+document.getElementById("subject").onchange = function() {subjectInputChange()};
+document.getElementById("msg").onchange = function() {msgInputChange()};
+
+// functions to change input/text area color and background color based on userinput
+function nameInputBackgroundChange() {
+  var nameField = document.getElementById("name");
+  nameField.style.backgroundColor = "white";
+  nameField.style.color = "#4D766E";
+}
+
+function nameInputChange() {
+  var nameField = document.getElementById("name");
+  if (nameField.value === '') {
+    nameField.style.backgroundColor = "#4D766E";
+    nameField.style.color = "white";
+  }
+}
+
+function mailInputBackgroundChange() {
```

Comment:
## Hm...also not very DRY.

### The Problem:

You have six more functions from here down, and they all do essentially the same thing...

### Challenge:

I bet you can refactor the `handleNameInputField` event handling function I pseudo-coded for you in the previous comment to be more general, so that it can handle the input field changes **for all four input fields in your form.**

**Hint:** Usually, when you want to take several highly specific functions and turn them into a single, more generalized function, the number of inputs to the function *increases*.